schematic/store/synapse.py

```diff
@@ -1517,6 +1520,59 @@ async def store_async_annotation(self, annotation_dict: dict) -> Annotations:
         )
         return await annotation_class.store_async(synapse_client=self.syn)
 
+    def process_row_annotations(
+        self,
+        dmge: DataModelGraphExplorer,
+        metadataSyn: dict,
+        hideBlanks: bool,
+        csv_list_regex: str,
+        annos: dict,
+        annotation_keys: str,
+    ):
+        """processes metadata annotations
+
+        Args:
+            dmge (DataModelGraphExplorer): data model graph explorer
+            metadataSyn (dict): metadata used for Synapse storage
+            hideBlanks (bool): if true, does not upload annotation keys with blank values.
+            csv_list_regex (str):
+            annos (dict):
+            annotation_keys (str): display_label/class_label
+
+        Returns:
+            dict: annotations as a dictionary
+        """
+        for anno_k, anno_v in metadataSyn.items():
+            # Remove keys with nan or empty string values from dict of annotations to be uploaded
+            # if present on current data annotation
+            if hideBlanks and (
+                (isinstance(anno_v, str) and anno_v.strip() == "")
```

Comment:
this is new in comparison with the old one. Here I added functionality to handle long white space